docs/api.md

```diff
@@ -1273,7 +1273,7 @@ or updates existing device details for this user/session
 If no device `id` is specified,
 both `name` and `type` must be provided.
 If a device `id` is specified,
-at least one of `name`, `type`, `pushCallback` and `pushPublicKey`
+at least one of `name`, `type`, `pushCallback` or the couple `pushPublicKey` and `pushAuthKey`
 must be present.
```

Comment:
FYI: The endpoint (pushCallback), p256dh (pushPublicKey) and auth (pushAuthKey) are all fairly tightly connected, particularly if you're going to be using data. You may wish to reflect that here so that users are not caught off-guard if, say the pushCallback changes triggering new values for pushPublicKey and pushAuthKey.